contracts/actions/flashloan/FLAaveV3CarryDebt.sol

```diff
@@ -0,0 +1,149 @@
+// SPDX-License-Identifier: MIT
+pragma solidity =0.8.10;
+
+import "../ActionBase.sol";
+import "../../interfaces/IDSProxy.sol";
+import "../../interfaces/IFLParamGetter.sol";
+import "../../interfaces/aaveV3/IPoolV3.sol";
+import "../../core/strategy/StrategyModel.sol";
+import "../../utils/TokenUtils.sol";
+import "../../utils/ReentrancyGuard.sol";
+import "./helpers/FLHelper.sol";
+import "../../interfaces/flashloan/IFlashLoanBase.sol";
+import "../../interfaces/aaveV3/IDebtToken.sol";
+
+/// @title Action that gets and receives a FL from Aave V3 and does not return funds but opens debt position on Aave V3
+contract FLAaveV3CarryDebt is ActionBase, StrategyModel, ReentrancyGuard, FLHelper, IFlashLoanBase {
+    using SafeERC20 for IERC20;
+    using TokenUtils for address;
+    //Caller not aave pool
+    error OnlyAaveCallerError();
+    //FL Taker must be this contract
+    error SameCallerError();
+    //Interest rate can't be 0 for opening depth position on Aave V3
+    error NoInterestRateSetError();
+    //Credit delegation allowance must be 0 after FL
+    error CreditDelegationAllowanceLeftError();
+
+    bytes4 constant RECIPE_EXECUTOR_ID = bytes4(keccak256("RecipeExecutor"));
+
+    /// @inheritdoc ActionBase
+    function executeAction(
+        bytes memory _callData,
+        bytes32[] memory,
+        uint8[] memory,
+        bytes32[] memory
+    ) public override payable returns (bytes32) {
+        FlashLoanParams memory flData = parseInputs(_callData);
+
+        // check to make sure modes are not 0
+        for (uint256 i = 0; i < flData.modes.length; i++) {
```

Comment:
More of a pet peeve but be consistent with the pre/post increment in a loop, `++i` should be more gas optimal in 8.10 version